sherpa/image/DS9.py

```diff
@@ -454,11 +450,26 @@ def doOpen(self):
         if self.isOpen():
             return
 
-        _Popen(
+        # We want to fork ds9. This is possible with os.fork, but
+        # it doesn't work on Windows. At present Sherpa does not
+        # run on Windows, so it is not a serious problem, but it is
+        # not clear if it is an acceptable, or sensible, option.
+        #
+        p = _Popen(
             args=('ds9', '-title', self.template, '-port', "0"),
             cwd=None,
+            close_fds=True, stdin=None, stdout=None, stderr=None
         )
 
+        # Trick to stop a ResourceWarning warning to be created when
+        # running sherpa/tests/test_image.py
+        #
+        # Adapted from https://hg.python.org/cpython/rev/72946937536e
+        # but I am completely mis-using it, so it is unclear how
+        # sensible a hack this is.
+        #
+        p.returncode = 0
```

Comment:
I believe this statement to be more appropriate inside the `if self.isOpen()` condition. It's still a slight hack, but at least it's more semantically appropriate to say that the return code is 0 when things are OK, and leave it undefined when they are not, e.g. when the timeout is reached and the window is still not open (if I interpret the code correctly).

I tested this in a commit I am going to push for simplicity. If @DougBurke disagrees I can always revert back to the original code.